### ISSUE TITLE
executables not runnable when compiled on OSX 10.15 (Catalina)

### ISSUE THREAD

**Issue by WebFreak001** [ldc-developers/ldc#3227]
See https://github.com/Pure-D/code-d/issues/269

Temporarily it was resolved with GitHub Actions as the CI building the binaries on an older OSX version which produced working executables for future OSX versions. However now GitHub Actions has upgraded to OSX 10.15 so the executables it generates no longer work.

Example run on CI: https://github.com/Pure-D/serve-d/runs/302180066 (check the Run tests tab)

**Comment by kinke** [comment]
I highly doubt this is compiler-specific, even less so LDC-specific (as the OP of the serve-d issue uses DMD 2.088). It sounds more like an issue wrt. a symbolic .dylib link resolving to a newer incompatible version when linking on 10.15, a version with possibly breaking API changes or so. That can be identified by comparing the .dylib dependencies of the produced executables. - Not sure if that would fix things, but do you use the `MACOSX_DEPLOYMENT_TARGET` env variable?

**Comment by WebFreak001** [comment]
you are right, this happens with dmd too. I tried MACOSX_DEPLOYMENT_TARGET=10.12 now but it still crashes like before.

I tried registering signal handlers for SIGABRT and others in a `shared static this()`

The executable still crashes with error code -6 even with registering error handlers right at the start as it didn't even execute them. Because of this I think this might be a phobos bug then?
